cli/tests/test_split_command.rs

```diff
@@ -1142,6 +1142,240 @@ fn test_split_with_message() {
     ");
 }
 
+#[test]
+fn test_split_move_first_commit() {
+    let test_env = TestEnvironment::default();
+    test_env.run_jj_in(".", ["git", "init", "repo"]).success();
+    let work_dir = test_env.work_dir("repo");
+
+    work_dir.write_file("file1", "foo\n");
+    work_dir.write_file("file2", "bar\n");
+    work_dir.run_jj(["commit", "-m", "file2"]).success();
+    work_dir.write_file("file3", "bar\n");
+    work_dir.run_jj(["commit", "-m", "file3"]).success();
+    work_dir.write_file("file4", "bar\n");
+    work_dir.run_jj(["commit", "-m", "file4"]).success();
+    work_dir.run_jj(["new", "root()"]).success();
+    work_dir.write_file("file5", "bar\n");
+    work_dir.run_jj(["commit", "-m", "file5"]).success();
+
+    insta::assert_snapshot!(get_log_output(&work_dir), @r"
+    @  royxmykxtrkr true
+    ○  mzvwutvlkqwt false file5
+    │ ○  kkmpptxzrspx false file4
+    │ ○  rlvkpnrzqnoo false file3
+    │ ○  qpvuntsmwlqt false file2
+    ├─╯
+    ◆  zzzzzzzzzzzz true
+    [EOF]
+    ");
+
+    // insert the commit before the source commit
+    let output = work_dir.run_jj([
+        "split",
+        "-m",
+        "file1",
+        "-r",
+        "qpvuntsmwlqt",
+        "--insert-before",
+        "qpvuntsmwlqt",
+        "file1",
+    ]);
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Rebased 2 descendant commits
+    First part: vruxwmqv bf94c29a file1
+    Second part: qpvuntsm 66b1d4f1 file2
+    [EOF]
+    ");
+
+    insta::assert_snapshot!(get_log_output(&work_dir), @r"
+    @  royxmykxtrkr true
+    ○  mzvwutvlkqwt false file5
+    │ ○  kkmpptxzrspx false file4
+    │ ○  rlvkpnrzqnoo false file3
+    │ ○  qpvuntsmwlqt false file2
+    │ ○  vruxwmqvtpmx false file1
+    ├─╯
+    ◆  zzzzzzzzzzzz true
+    [EOF]
+    ");
+
```

Comment:
I think it's better to assert content of the split commits. Maybe we can include real file names in the log output?